plugins/repository-s3/src/test/java/org/opensearch/repositories/s3/S3BlobStoreContainerTests.java

```diff
@@ -916,6 +916,15 @@ public void testListBlobsByPrefixInLexicographicOrderWithLimitLessThanPageSize()
         testListBlobsByPrefixInLexicographicOrder(2, 1, BlobContainer.BlobNameSortOrder.LEXICOGRAPHIC);
     }
 
+    /**
+     * Test the boundary value at page size to ensure
+     * unnecessary calls are not made to S3 by fetching the next page.
+     * @throws IOException
+     */
+    public void testListBlobsByPrefixInLexicographicOrderWithLimitEqualToPageSize() throws IOException {
+        testListBlobsByPrefixInLexicographicOrder(5, 1, BlobContainer.BlobNameSortOrder.LEXICOGRAPHIC);
```

Comment:
This test fails without the change, right?

Reply:
Yes (it fetches 2 pages instead of 1)